### PR TITLE
Add script to convert CSV to JSON for mongo import

### PR DESCRIPTION
**Pull request by eldeal**
### What
-Separated code lists into individual files for easier management
-Added indexing of codelist ID field to separate from mongo _id field
-Created and documented a script to allow converting a CSV file to additional JSON lines for mongo import

### How to review

- Checkout branch
- Find a V4 file with an unimported dimension
- Follow documentation to import the file
- Check documented outputs occur

### Who can review
@Matt-Guest - I probably need to discuss some of these changes with you as I have made very slight changes to mongo structures
 @robchamberspfc - If you want to check this branch out and tell me if the documentation is clear enough/things seem to work?

### CODE REVIEW

File: scripts/add.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		log.Fatal("failed to append code file to setup script", err)
+	}
+}
+
+func createCodeList(header []string) string {
+	listID := uuid.New().String()
+
+	// if header does not include an ID we might not be able to create one here
+	// this might need to be done in the recipe API and then passed in via csv
+	if len(header[0]) > 0 {
```
Comment (eldeal) [inline scripts/add.go]: needs updating to check if the value after _ is 'codelist'

File: scripts/setup.sh [hunk 2]
```diff
@@ -2,10 +2,22 @@
...
 
 scriptDir=$( dirname "${BASH_SOURCE[0]}" )
-mongoimport --db codelists --collection codelists --file $scriptDir/codelists.json
-mongoimport --db codelists --collection codes --file $scriptDir/codes.json
+cd $scriptDir || exit 2
+
+import_to() {
+        local collection=$1 file=$2; shift 2
+        echo Importing $file ...
+        mongoimport --db codelists --collection $collection --file $file || exit 2
```
Comment (Matt-Guest) [inline scripts/setup.sh]: It maybe worth allowing the mongodb URI to be configured here

File: scripts/add.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+	return listID
+}
+
+func appendToFile(b []byte, filename string) error {
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
```
Comment (Matt-Guest) [inline scripts/add.go]: Is this being closed twice? here and at line 187?

File: scripts/add.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	for _, v := range recs {
+		id := uuid.New().String()
+		code := Code{
+			ID:    id,
+			Code:  v[0],
+			Label: v[1],
+			Links: models.CodeLinks{
+				CodeList: models.Link{
+					ID:   listID,
+					Href: "http://localhost:22400/code-lists/" + listID,
```
Comment (Matt-Guest) [inline scripts/add.go]: Do we still want to hardcode the hostname here?

File: scripts/add.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+			}
+
+			jsonLineChan <- b
+		}(code)
+	}
+
+	wg.Wait()
+	close(jsonLineChan)
+
+	if err := f.Close(); err != nil {
```
Comment (Matt-Guest) [inline scripts/add.go]: Is this being closed twice via a `defer` at line 75

File: scripts/add.go (hunk 4 above)
Comment (eldeal) [reply scripts/add.go]: all the links in the environments use localhost to refer to themselves atm. we may need to revisit that wider but for our purposes right now it should work.

File: scripts/add.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+		log.Fatal("failed to append code file to setup script", err)
+	}
+}
+
+func createCodeList(header []string) string {
+	listID := uuid.New().String()
+
+	// if header does not include an ID we might not be able to create one here
+	// this might need to be done in the recipe API and then passed in via csv
+	parts := strings.Split(header[0], "_")
```
[the remark this answers is not in the document]
Comment (eldeal) [reply scripts/add.go]: have simplified it so the input file should just contain the codelistID in the header, removing this split. updated the doc to explain what the header should look like. if an ID isn't provided we generate a new one, this is intended.

File: scripts/add.go (hunk 5 above)
Comment (eldeal) [reply scripts/add.go]: @gedge - if i change the enclosing function to return an error, and in the calling function log.Fatal(err) would this remove the need to check for close here? Not sure if it reintroduces the race condition

File: scripts/setup.sh (hunk 2 above)
Comment (eldeal) [reply scripts/setup.sh]: maybe a feature worth adding when it's needed? dont know that it's needed right now

File: scripts/add.go (hunk 5 above)
Comment (gedge) [reply scripts/add.go]: it won't introduce a race condition.

I would remove this `f.Close()` stanza - but it looks like errors in deferred function calls are ignored, though!  Consider using something like the `Close` func at the end of: http://www.blevesearch.com/news/Deferred-Cleanup,-Checking-Errors,-and-Potential-Problems/

Comment (eldeal) [reply scripts/add.go]: well thats less than ideal. will do!

**Pull request merged**